client/landing/gutenboarding/onboarding-block/signup-form/index.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+import React, { useState } from 'react';
+import { Button, ExternalLink, TextControl, Modal } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __experimentalCreateInterpolateElement } from '@wordpress/element';
+import { __ as NO__, _x as NO_x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { USER_STORE } from '../../stores/user';
+import './style.scss';
+
+const SignupForm = () => {
+	const [ emailVal, setEmailVal ] = useState( '' );
+	const { createAccount } = useDispatch( USER_STORE );
+	const isFetchingNewUser = useSelect( select => select( USER_STORE ).isFetchingNewUser() );
+	const newUser = useSelect( select => select( USER_STORE ).getNewUser() );
+	const newUserError = useSelect( select => select( USER_STORE ).getNewUserError() );
+
+	const handleSignUp = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		createAccount( { email: emailVal, is_passwordless: true, signup_flow_name: 'gutenboarding' } );
+	};
+
+	return (
+		<Modal
+			className="signup-form"
+			isDismissible={ false }
+			title={ NO__( 'Sign up to save your changes' ) }
+			onRequestClose={ noop }
+		>
+			<form onSubmit={ handleSignUp }>
+				<TextControl
+					label={ NO__( 'Your Email Address' ) }
+					value={ emailVal }
+					disabled={ isFetchingNewUser }
+					type="email"
+					onChange={ setEmailVal }
+					placeholder={ NO_x(
+						'E.g., yourname@email.com',
+						"An example of a person's email, use something appropriate for the locale"
+					) }
+				/>
+				<div className="signup-form__footer">
+					<p className="signup-form__terms-of-service-link">{ renderTos() }</p>
+
+					<Button
+						type="submit"
+						className="signup-form__submit"
+						disabled={ isFetchingNewUser }
```

Comment:
Could we also disable this when `! emailVal` 

Something similar to what we do in https://github.com/Automattic/wp-calypso/blob/c9e127d/client/blocks/signup-form/passwordless.jsx

Reply:
I personally I like setting `type=email` on the input box and then letting default browser behaviour kick in, that way we know the error message will be accessible. Usually I think the browser won't tell you the email is invalid until you attempt to submit.

Maybe it can be in another PR after we have more of an idea of how passwordless signup should look and feel in gutenboarding. Will ping @Automattic/gutenboarding-design so they can weigh in/think about how invalid user input should work in this brave new world.